packages/vm/src/instance.rs

```diff
@@ -29,6 +31,7 @@ static WASM_PAGE_SIZE: u64 = 64 * 1024;
 pub struct Instance<S: Storage + 'static, A: Api + 'static, Q: Querier + 'static> {
     wasmer_instance: wasmer_runtime_core::instance::Instance,
     pub api: A,
+    pub required_features: HashSet<String>,
```

Comment:
Ah, just saw the instance itself contains this info.
This is a great design! Super flexible where we return the info.